cfn-resources/private-endpoint-regional-mode/cmd/resource/resource.go

```diff
@@ -149,25 +137,25 @@ func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.P
 	return handler.ProgressEvent{}, errors.New("not implemented: List")
 }
 
-func resourcePrivateEndpointRegionalModeUpdate(currentModel *Model, client *mongodbatlas.Client, enabled bool) (handler.ProgressEvent, error) {
-	regionalizedPrivateEndpointSetting, response, err := client.PrivateEndpoints.UpdateRegionalizedPrivateEndpointSetting(context.Background(), *currentModel.ProjectId, enabled)
+func resourcePrivateEndpointRegionalModeUpdate(currentModel *Model, client *util.MongoDBClient, enabled bool) (handler.ProgressEvent, error) {
+	_, response, err := client.AtlasV2.PrivateEndpointServicesApi.ToggleRegionalizedPrivateEndpointSetting(context.Background(), *currentModel.ProjectId,
+		&admin.ProjectSettingItem{
+			Enabled: enabled,
+		}).Execute()
 	if err != nil {
-		return progress_events.GetFailedEventByResponse(
-			fmt.Sprintf("Error in enabling regionalized settings : %s", err.Error()),
-			response.Response), nil
+		return progressevent.GetFailedEventByResponse(err.Error(), response), nil
 	}
 
-	// Response
 	return handler.ProgressEvent{
 		OperationStatus: handler.Success,
 		Message:         "Create Complete",
-		ResourceModel:   regionalPrivateEndpointToModel(*currentModel, regionalizedPrivateEndpointSetting),
+		ResourceModel:   regionalPrivateEndpointToModel(*currentModel),
 	}, nil
 }
 
-func isRegModeSettingExists(currentModel *Model, client *mongodbatlas.Client) bool {
+func isRegModeSettingExists(currentModel *Model, client *util.MongoDBClient) bool {
 	var isExists bool
```

Comment:
I see the logic with isExists was already there, but I think not having that variable will make the function implementation cleaner and easier to understand

Reply:
updated